src/components/DataTable/EditCellField.js

```diff
@@ -0,0 +1,153 @@
+import cx from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../Icon';
+import EditCellActions from './EditCellActions';
+
+export default class EditCellField extends React.Component {
+  static propTypes = {
+    /**
+     * Specify whether the Cell Field is disabled
+     */
+    disabled: PropTypes.bool,
+    /**
+     * Pass in an error that is associated with the edit status of the cell
+     * field
+     */
+    error: PropTypes.string,
+
+    /**
+     * The id used to uniquely identify the underlying input node
+     */
+    id: PropTypes.string.isRequired,
+
+    /**
+     * Specify whether the Edit Cell is currently being saved, or not
+     */
+    isSaving: PropTypes.bool,
+
+    /**
+     * Pass in the label text that is used to describe this cell to screen
+     * readers
+     */
+    labelText: PropTypes.string.isRequired,
+
+    /**
+     * Specify a hook for when the user cancels their current edit flow
+     */
+    onCancel: PropTypes.func.isRequired,
+
+    /**
+     * Specify a hook to listen into whenever the underlying input node changes
+     */
+    onChange: PropTypes.func.isRequired,
+
+    /**
+     * Specify a hook for when the user saves their current edit flow
+     */
+    onSave: PropTypes.func.isRequired,
+
+    /**
+     * Specify the type of the Cell Field. For now, only supports text. In the
+     * future this should support a variety of input types.
+     */
+    type: PropTypes.oneOf(['text']).isRequired,
+
+    /**
+     * The value of the Cell
+     */
+    value: PropTypes.string.isRequired,
+  };
+
+  state = {
+    isEditing: true,
+  };
+
+  componentDidMount() {
+    // Set cursor position to end of control
+    this.inputNode.value = '';
+    this.inputNode.value = this.props.value;
```

Comment:
Why two assignment here…?

Reply:
Will actually reset the input to have the correct cursor position at the end of the string.